packages/legend-query-builder/src/components/__tests__/QueryBuilderVariables.test.tsx

```diff
@@ -0,0 +1,228 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { test, expect } from '@jest/globals';
+import {
+  waitFor,
+  fireEvent,
+  getByTitle,
+  act,
+  getByText,
+} from '@testing-library/react';
+import { TEST_DATA__simpleProjection } from '../../stores/__tests__/TEST_DATA__QueryBuilder_Generic.js';
+import { TEST_DATA__ModelCoverageAnalysisResult_ComplexRelational } from '../../stores/__tests__/TEST_DATA__ModelCoverageAnalysisResult.js';
+import TEST_DATA__ComplexRelationalModel from '../../stores/__tests__/TEST_DATA__QueryBuilder_Model_ComplexRelational.json';
+import { integrationTest } from '@finos/legend-shared';
+import {
+  create_RawLambda,
+  PrimitiveType,
+  stub_RawLambda,
+} from '@finos/legend-graph';
+import { QUERY_BUILDER_TEST_ID } from '../QueryBuilder_TestID.js';
+import { TEST__setUpQueryBuilder } from '../QueryBuilderComponentTestUtils.js';
+import { CUSTOM_DATE_PICKER_OPTION } from '../shared/CustomDatePicker.js';
+import { variableExpression_setName } from '../../stores/shared/ValueSpecificationModifierHelper.js';
+
+test(
+  integrationTest(
+    'Query builder shows validation error for parameter name if existing duplicate constant name',
+  ),
+  async () => {
+    const { renderResult, queryBuilderState } = await TEST__setUpQueryBuilder(
+      TEST_DATA__ComplexRelationalModel,
+      stub_RawLambda(),
+      'model::relational::tests::simpleRelationalMapping',
+      'model::MyRuntime',
+      TEST_DATA__ModelCoverageAnalysisResult_ComplexRelational,
+    );
+    await act(async () => {
+      queryBuilderState.initializeWithQuery(
+        create_RawLambda(
+          TEST_DATA__simpleProjection.parameters,
+          TEST_DATA__simpleProjection.body,
+        ),
+      );
+      queryBuilderState.setShowParametersPanel(true);
+      queryBuilderState.constantState.setShowConstantPanel(true);
+    });
+
+    await waitFor(() =>
+      renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_CONSTANTS),
+    );
+    const constantsPanel = renderResult.getByTestId(
+      QUERY_BUILDER_TEST_ID.QUERY_BUILDER_CONSTANTS,
+    );
+    fireEvent.click(getByTitle(constantsPanel, 'Add Constant'));
+    await act(async () => {
+      if (!queryBuilderState.constantState.selectedConstant) {
+        return;
+      }
+      const varExpression =
+        queryBuilderState.constantState.selectedConstant.variable;
+      variableExpression_setName(varExpression, 'var_1');
+    });
+    await waitFor(() => renderResult.getByText('Create'));
+    fireEvent.click(renderResult.getByText('Create'));
+
+    await waitFor(() =>
+      renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_PARAMETERS),
+    );
+    const parametersPanel = renderResult.getByTestId(
+      QUERY_BUILDER_TEST_ID.QUERY_BUILDER_PARAMETERS,
+    );
+
+    fireEvent.click(getByTitle(parametersPanel, 'Add Parameter'));
+
+    expect(
```

Comment:
this is actually a bug. when we generate the name for parameters we check existing names to determine what we generate. We probably don't check include constants in this check. Could we fix that. and then for this test we should write out the duplicate and see the error

Reply:
I think we don't check because we don't need to since when we call `generateEnumerableNameFromToken` for constant our prefix we put in front is `c_var` while for parameters it's `var` so there is never a conflict between c_var_1,etc. and var_1, etc. That's why I changed the constant name directly to be var_1 to cause this error